### ISSUE TITLE
Lags

### ISSUE THREAD

**Issue by CarterSnich** [thonny/thonny#1197]
Laggy when typing, also the auto complete dialogue box or something does not pops up immediately.

**Comment by CarterSnich** [comment]
It's not laggy tho when I'm typing on the buit-in Shell.

**Comment by CarterSnich** [comment]
EDIT: the lag only occurs when the codes gets many, but my code right now has only 230 lines.

**Comment by aivarannamaa** [comment]
Auto complete is expected to be slow first time, but typing in a 230-line program should be fast. What is your platform?

**Comment by CarterSnich** [comment]
Windows 10 Pro x64

**Comment by lurch** [comment]
Try fiddling with the syntax-highlighting options to see if that improves things?

**Comment by aivarannamaa** [comment]
> Try fiddling with the syntax-highlighting options to see if that improves things?

Good idea! This means "Tools => Options => Editor". Try switching off optional features. If you find the one that causes the lag, please let me know as well.

**Comment by BlenderChrisDW** [comment]
I have noticed the same lag. I have found that using 'Highlight matching names' or 'Highlight local variables' causes the lag. All other options seem to work fine.

**Comment by CarterSnich** [comment]
The performance improved, I disabled both 'Highlighting matching names' and 'Highlight local variables'. Thanks for the tip. But, I think both features should not be disabled, 'coz it seems like it's been a tradition for every editor/IDE to have such features enabled by default. I'll just disable it tho.

About the auto-complete dialog box, yeah it was slow for the first time.

Sorry for the late response. Thanks again for the help.
